mi/dataset/parser/glider.py

```diff
@@ -1152,3 +1165,104 @@ def handle_metadata_particle(self, timestamp):
+    def _lacks_interpolate_entry_positions(self):
+        return self._start_gps is None or self._end_gps is None
+
+    def _put_value(self, glider_values, value, key):
+        for glider_value_entry in glider_values:
+            glider_value_id = glider_value_entry['value_id']
+            if glider_value_id == key:
+                glider_value_entry['value'] = value
+                break
+
+    def _set_start_and_end_gps_position_entries(self):
+        # Iterate the buffer to pinpoint entries with the first/last GPS points
+        for idx, glider_gps_position in enumerate(self._glider_gps_position_buffer):
+            if self._has_gps_positioning(glider_gps_position):
+                # only populate the end position if the start has previously populated
+                # as the interpolation only makes sense and works if they are different
+                if self._start_gps is None:
+                    self._start_gps = idx
+                else:
+                    self._end_gps = idx
+
+    def append_to_buffer(self, glider_gps_position):
+        self._glider_gps_position_buffer.append(glider_gps_position)
+
+    def get_size(self):
+        return len(self._glider_gps_position_buffer)
+
+    # Interpolate the buffered objects, then return them
+    def process_and_get_objects(self):
+        self._set_start_and_end_gps_position_entries()
+        # if the set of entries lacks gps lat/lon data, return it as-is
+        if self._lacks_interpolate_entry_positions():
+            return self._glider_gps_position_buffer
+
+        # Iterate the buffer to extract the values for interpolation
+        for idx, glider_gps_position in enumerate(self._glider_gps_position_buffer):
+            if self._start_gps <= idx <= self._end_gps:
+                glider_time = self._get_time(glider_gps_position.contents)
+                self.particle_ts.append(glider_time)
+
+                # If the lat,lon can be used for interpolation, capture them
+                if self._has_gps_positioning(glider_gps_position):
+                    self.gps_lat.append(self._get_value(glider_gps_position._values, GpsPositionParticleKey.M_GPS_LAT))
+                    self.gps_lon.append(self._get_value(glider_gps_position._values, GpsPositionParticleKey.M_GPS_LON))
+                    self.gps_time.append(glider_time)
+
+        # Set up the interpolation function for gps_lat
+        interp_function = interpolate.interp1d(self.gps_time, self.gps_lat, kind='linear', axis=0, copy=False)
+        # Use the particle time array to get the interpolated gps lat array
+        interpolated_gps_lat = interp_function(self.particle_ts)
+
+        # Set up the interpolation function for gps_lon
+        interp_function = interpolate.interp1d(self.gps_time, self.gps_lon, kind='linear', axis=0, copy=False)
+        # Use the particle time array to get the interpolated gps lon array
+        interpolated_gps_lon = interp_function(self.particle_ts)
+
```

Comment:
It looks like we are interpolating the values over a subset of the time range from the first gps time to the last gps time. I assume that this is the desired result but if we wanted to populate the gps lat/lon for all time values, I believe we could pass fill_value="extrapolate" as an arg to the spicy.interp1d function to extrapolate the gps lat/lon for particle time values outside of the actual gps time range.

Reply:
I added the extrapolation to glider.py. Thanks for your assistance!